src/profiler.cpp

```diff
@@ -614,6 +657,7 @@ void Profiler::fillFrameTypes(ASGCT_CallFrame* frames, int num_frames, NMethod*
 
 u64 Profiler::recordSample(void* ucontext, u64 counter, EventType event_type, Event* event) {
     atomicInc(_total_samples);
+    incrementSampleCounter();
```

Comment:
I'm thinking where the right place to increment the counter will be.
The counter is only relevant with JFR output, isn't it? Then `recordEvent` (or the places where it is called) seems a better alternative. Or can you imagine a different use case?
Note that some recorded events (e.g. `free` calls in native memory profiling mode) bypass `Profiler::recordSample`.